lib/Logger.ts

```diff
@@ -1,241 +1,179 @@
 	 * Temporary storage for timers.
 	 */
-	private static timers = new Map<string, number>();
+	private timers = new Map<string, number>();
 
 	/**
-	 * Temporary storage for log tags.
+	 * Temporary storage for timeouts.
 	 */
-	private static tags?: string[];
+	private timeouts = new Map<string, NodeJS.Timeout>();
 
 	/**
-	 * Stores the Winston Logger instance.
+	 * Stores the logger wrapper being used.
 	 */
-	protected logger: WinstonLogger;
+	protected logger: LogWrapper;
 
 	/**
-	 * Stores the transports for Winston.
+	 * Stores the HTTP server used to issue commands for the logger.
 	 */
-	protected transports: Transport[] = [];
-
-	/**
-	 * Default log level for all transports.
-	 */
-	protected level: string;
-
-	/**
-	 * Stores the timer for set-level command timeout option.
-	 */
-	private timer: NodeJS.Timeout | void;
+	public server: Server | void;
 
 	/**
 	 * Logger module class constructor.
 	 *
-	 * @param  config  Logger module configuration options.
+	 * @param   opts         Logger module configuration options.
+	 * @param   opts.tag     Tag with which the logger is being created.
+	 * @param   opts.client  Underlying abstract logger to override console.
 	 */
-	private constructor(config: LoggerConfigOptions) {
-		let setup: { [key: string]: unknown };
-
-		this.level = config.level.toLowerCase(); // Default
-
-		if (includes(config.targets, 'stdout')) {
-			setup = {
-				level: this.level
-			};
-
-			if (config.stdout && config.stdout.output === 'json') {
-				setup.format = Json(config.app, color());
-			} else {
-				setup.format = Text(config.app, color());
+	public constructor(opts: { tag?: string; client?: AbstractLogger } = {}) {
+		this.logger = getLogWrapper(output(), opts.client ?? console, opts.tag);
+		this.configure(level());
+
+		this.server = setupLogServer(...host(), this);
+
+		registerEvent(
+			this,
+			'ozlogger.http.changeLevel',
+			(data: {
+				event: string;
+				level: keyof typeof LogLevels;
+				duration: number;
+			}) => {
+				const newLevel = this.configure(data.level).toUpperCase();
+				this.warn(`Changed log level to ${newLevel}`);
+
+				this.schedule(
+					data.event,
+					() => {
+						const oldLevel = this.configure(level()).toUpperCase();
+						this.warn(`Reset log level to ${oldLevel}`);
+					},
+					data.duration
+				);
 			}
+		);
+	}
 
-			this.transports.push(new transports.Console(setup));
-		}
-
-		if (includes(config.targets, 'file')) {
-			setup = {
-				level: this.level,
-				decolorize: true
-			};
-
-			if (config.file) {
-				setup.filename = config.file.filename;
-				setup.format =
-					config.file.output === 'json'
-						? Json(config.app)
-						: Text(config.app);
-
-				if (config.file.maxsize) setup.maxsize = config.file.maxsize;
-			} else {
-				setup.filename = `${config.app}.log`;
-				setup.format = Text(config.app, color());
-			}
+	/**
+	 * Method for stopping HTTP server and cleaning up timeouts.
+	 */
+	public async stop(): Promise<void> {
```

Comment:
Yes. this is being used so that we can have integration tests with jest and handle the HTTP server teardown.

Reply:
I think we should use Simon to add this kind of method, as it is only used by tests. This will avoid "dead-code". Not a big problem for now, just a comment.